extensions/src/main/java/org/apache/falcon/extensions/ExtensionBuilder.java

```diff
@@ -19,14 +19,39 @@
 package org.apache.falcon.extensions;
 
 import org.apache.falcon.FalconException;
+import org.apache.falcon.Pair;
 import org.apache.falcon.entity.v0.Entity;
+import org.apache.falcon.entity.v0.feed.Schema;
 
+import java.io.InputStream;
 import java.util.Properties;
 import java.util.List;
 
 /**
  * Extension interface to be implemented by all trusted and custom extensions.
  */
 public interface ExtensionBuilder {
-    List<Entity> getEntities(final String extensionName, final Properties extensionProperties) throws FalconException;
+
+    /**
+     * @param extensionName extension name.
+     * @param extensionConfigStream stream comprising of the extension properties.
+     * @return List of the entities that are involved in the extension.
+     * @throws FalconException
+     */
+    List<Entity> getEntities(final String extensionName, final InputStream extensionConfigStream)
+        throws FalconException;
+
+    /**
+     * @param extensionName extension name.
+     * @param extensionConfigStream Properties supplied will be validated.
+     * @throws FalconException
+     */
+    void validateExtension(final String extensionName, final InputStream extensionConfigStream) throws FalconException;
```

Comment:
Apologies for not pointing this out earlier, validateExtension is misleading. We are actually validating the configuration supplied by the user. Please change accordingly.